From 7b82d37c3b5b46b01b80ced412324ac34ade6191 Mon Sep 17 00:00:00 2001
From: Wang Yufen <wangyufen@huawei.com>
Date: Wed, 11 Jan 2023 10:10:50 +0800
Subject: [PATCH 1382/1398] [Huawei] kabi: add early_demux_handler and
 early_demux back to fix kabi broken in struct net_protocol and inet6_protocol

Offering: HULK
hulk inclusion
category: bugfix
bugzilla: 188217

--------------------------------

Add early_demux_handler and early_demux back to fix kabi broken in
struct net_protocol and inet6_protocol.

Signed-off-by: Wang Yufen <wangyufen@huawei.com>
---
 include/net/protocol.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/include/net/protocol.h b/include/net/protocol.h
index 0fd2df844fc7..2b778e1d2d8f 100644
--- a/include/net/protocol.h
+++ b/include/net/protocol.h
@@ -35,6 +35,8 @@
 
 /* This is used to register protocols. */
 struct net_protocol {
+	int			(*early_demux)(struct sk_buff *skb);
+	int			(*early_demux_handler)(struct sk_buff *skb);
 	int			(*handler)(struct sk_buff *skb);
 
 	/* This returns an error if we weren't able to handle the error. */
@@ -51,6 +53,8 @@ struct net_protocol {
 
 #if IS_ENABLED(CONFIG_IPV6)
 struct inet6_protocol {
+	void	(*early_demux)(struct sk_buff *skb);
+	void    (*early_demux_handler)(struct sk_buff *skb);
 	int	(*handler)(struct sk_buff *skb);
 
 	/* This returns an error if we weren't able to handle the error. */
-- 
2.17.1

